test/parallel/test-runner-mock-timers.js

```diff
@@ -232,6 +232,82 @@ describe('Mock Timers Test Suite', () => {
       });
     });
 
+    describe('setImmediate Suite', () => {
+      it('should keep setImmediate working if timers are disabled', (t, done) => {
+        const now = Date.now();
+        const timeout = 2;
+        const expected = () => now - timeout;
+        global.setImmediate(common.mustCall(() => {
+          assert.strictEqual(now - timeout, expected());
+          done();
+        }));
+      });
+
+      it('should work with the same params as the original setImmediate', (t) => {
+        t.mock.timers.enable(['setImmediate']);
+        const fn = t.mock.fn();
+        const args = ['a', 'b', 'c'];
+        global.setImmediate(fn, ...args);
+        t.mock.timers.tick(9999);
+
+        assert.strictEqual(fn.mock.callCount(), 1);
+        assert.deepStrictEqual(fn.mock.calls[0].arguments, args);
+      });
+
+      it('should not advance in time if clearImmediate was invoked', (t) => {
+        t.mock.timers.enable(['setImmediate']);
+
+        const fn = mock.fn();
```

Comment:
maybe this can be `common.mustNotCall()` instead of a mock + assertion?
As we don't need any of the other `mock` features, might be more explicit etc

Reply:
nice! just did it! o/